eZ/Publish/Core/FieldType/Author/Type.php

```diff
@@ -21,6 +22,23 @@
  */
 class Type extends FieldType
 {
+    /**
+     * Default value type current user.
+     */
+    const DEFAULT_CURRENT_USER = 1;
+
+    /**
+     * Default value empty.
+     */
+    const DEFAULT_EMPTY = -1;
```

Comment:
Any reason why you picked -1 over for instance 0? Other cases you found like that?
Asking as I can't right now remember us using negative ints much for int fields.

Reply:
I was having hard times to differentiate this value from this which I got from the database for existing FieldTypes (e.g. Article), as we wanted to have prefilling by default there.

Reply:
Ok, so you're not using 0 as todays default (Null) for data_int1. is casted to int in Storage engine so it becomes 0, which is causing your trouble to decide on value. Right?

Is 0 mapped to DEFAULT_CURRENT_USER somewhere for us btw, so editing existing data works as intended?

@alongosz SIDE: you see any issues with using negative int's for data_int1 column? Or should we rather keep the values in the positive range?

Reply:
>Ok, so you're not using 0 as todays default (Null) for data_int1. is casted to int in Storage engine so it becomes 0, which is causing your trouble to decide on value. Right?

Yes, that was my point.

>Is 0 mapped to DEFAULT_CURRENT_USER somewhere for us btw, so editing existing data works as intended?

So you are suggesting switching values? (0, both for prefilling chosen by editor and also for no value from db, 1 for no-prefilling option). It could be confusing, but probably would work.

Reply:
> So you are suggesting switching values?

not necessarily, before even suggesting that I was asking how the system behaves when db has null value. Both when it comes to content type view/edit and for content edit.

context: People upgrading would expect default value to just be current user somehow, so if code here handles that or if we are missing upgrade script where those upgrading can choose between setting these to 1 for user or -1/2 for todays null value.

Reply:
@andrerom I tested adding/editing content/content type with/without prefilling, everything seems to be working fine.